gtsam/linear/NoiseModel.cpp

```diff
@@ -311,10 +312,9 @@ void Diagonal::WhitenInPlace(Eigen::Block<Matrix> H) const {
 
 namespace internal {
 // switch precisions and invsigmas to finite value
-// TODO: why?? And, why not just ask s==0.0 below ?
 static void fix(const Vector& sigmas, Vector& precisions, Vector& invsigmas) {
   for (Vector::Index i = 0; i < sigmas.size(); ++i)
-    if (!std::isfinite(1. / sigmas[i])) {
+    if (sigmas[i] <= 1e-12) {
```

Comment:
Is there a reason for this change?

BTW we are in progress of reviewing the entire Constrained model...

Reply:
This was a TODO item and seemed cheaper and better than checking for infinite. Made sense to me and the change was easy so cleared that technical debt.

Reply:
I think the intention here is to explicitly check for 0.

Ping @dellaert

Reply:
Indeed, this is a behavior change. Please undo or replace with a better test that has the *same* behavior, and explain how.

Reply:
Wouldn't numerical stability require that we don't explicitly check for 0? A value like `6e-30` (which is considered 0) would not pass that check and that would explain why the original code was checking for infinity instead.

Reply:
Comments still stands as is. I have no time to engage in long dialog :-(